samples/components/dialects/Hive12Dialect.tdd

```diff
@@ -1100,106 +1158,174 @@
+      <argument type='localstr' />
+      <argument type='datetime' />
+      <argument type='datetime' />
+    </date-function>
+    <date-function name='DATEDIFF' return-type='int'>
+      <formula part='year'>CAST(YEAR(%2) - YEAR(%1) AS BIGINT)</formula>
+      <formula part='quarter'>CAST((4 * YEAR(%2) + CAST((MONTH(%2) - 1) / 3 + 1 AS BIGINT)) - (4 * YEAR(%1) + CAST((MONTH(%1) - 1) / 3 + 1 AS BIGINT)) AS BIGINT)</formula>
+      <formula part='month'>CAST((12 * YEAR(%2) + MONTH(%2)) - (12 * YEAR(%1) + MONTH(%1)) AS BIGINT)</formula>
+      <formula part='dayofyear'>DATEDIFF(TO_DATE(%2), TO_DATE(%1))</formula>
+      <formula part='day'>DATEDIFF(TO_DATE(%2), TO_DATE(%1))</formula>
+      <formula part='weekday'>DATEDIFF(TO_DATE(%2), TO_DATE(%1))</formula>
       <formula part='week'>FLOOR(DATEDIFF(NEXT_DAY(%2,&apos;SU&apos;),NEXT_DAY(%1,&apos;SU&apos;))/7)</formula>
+      <formula part='hour'>CAST((DATEDIFF(TO_DATE(%2), TO_DATE(%1)) * 24 + COALESCE(CAST(SUBSTR(%2,12,2) AS INT), 0) - COALESCE(CAST(SUBSTR(%1,12,2) AS INT), 0)) AS BIGINT)</formula>
+      <formula part='minute'>CAST(((DATEDIFF(TO_DATE(%2), TO_DATE(%1)) * 24 + COALESCE(CAST(SUBSTR(%2,12,2) AS INT), 0) - COALESCE(CAST(SUBSTR(%1,12,2) AS INT), 0)) * 60 + COALESCE(MINUTE(%2), 0) - COALESCE(MINUTE(%1), 0)) AS BIGINT)</formula>
+      <formula part='second'>CAST((((DATEDIFF(TO_DATE(%2), TO_DATE(%1)) * 24 + COALESCE(CAST(SUBSTR(%2,12,2) AS INT), 0) - COALESCE(CAST(SUBSTR(%1,12,2) AS INT), 0)) * 60 + COALESCE(MINUTE(%2), 0) - COALESCE(MINUTE(%1), 0)) * 60 + COALESCE(SECOND(%2), 0) - COALESCE(SECOND(%1), 0)) AS BIGINT)</formula>
       <argument type='localstr' />
       <argument type='datetime' />
       <argument type='datetime' />
       <argument type='localstr' />
     </date-function>
     <date-function name='DATENAME' return-type='str'>
-      <formula part='year'>CAST(YEAR(%2) AS STRING)</formula>
-      <formula part='quarter'>CAST(CAST((MONTH(%2) - 1) / 3 + 1 AS BIGINT) AS STRING)</formula>
-      <formula part='month'>CAST(MONTH(%2) AS STRING)</formula>
-      <formula part='dayofyear'>CAST(DATEDIFF(TO_DATE(%2), TO_DATE(CAST(TRUNC(%2,&apos;YY&apos;) AS DATE))) + 1 AS STRING)</formula>
-      <formula part='day'>CAST(DAY(%2) AS STRING)</formula>
-      <formula part='weekday'>CAST((8 + DATEDIFF(CAST(%2 AS DATE),NEXT_DAY(%2,&apos;SU&apos;))) AS STRING)</formula>
-      <formula part='week'>FORMAT_NUMBER(FLOOR((14 + DATEDIFF(%2, TRUNC(%2,&apos;YY&apos;)) + DATEDIFF(TRUNC(%2,&apos;YY&apos;),NEXT_DAY(TRUNC(%2,&apos;YY&apos;),&apos;SU&apos;)))/7),0)</formula>
-      <formula part='hour'>CAST(HOUR(%2) AS STRING)</formula>
-      <formula part='minute'>COALESCE(CAST(MINUTE(%2) AS STRING), &apos;&apos;)</formula>
-      <formula part='second'>COALESCE(CAST(SECOND(%2) AS STRING), &apos;&apos;)</formula>
+      <formula part='year'>CAST(YEAR(%1) AS STRING)</formula>
```

Comment:
`%1` is incorrect.  That is the string argument.  It seems like this was generated with the unit tests that outputs the dialects, which comes with a lot of caveats, see the notes in the feature/story.

The same comment basically applies to all of the changes to the various `date-function`.